Fix wrong display of multiline messages in the log after filtering

jason810496 opened the pull request:
closes: #41265

### Issue Context
After research, the issue is caused by `airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts`, the API just return the full raw logs.
Additionally, incorrect display behavior also occurs when filtering by file source.

### Main Fix Idea
The `currentLogLevel` and `currentFileSource` should remain same until a new `logLevel` or `fileSource` is encountered.

### Screenshot After Fix

**Without any filters**
<img width="1401" alt="Without any filters" src="https://github.com/user-attachments/assets/6297f0ce-1337-4185-b0c5-8c4585b8fbc8">
**Filter with level**
<img width="1402" alt="Filter with level" src="https://github.com/user-attachments/assets/70458852-25d3-461f-b5c7-c8665907bb7f">
**Filter with file source**
<img width="1340" alt="Filter with file source" src="https://github.com/user-attachments/assets/06f7a866-9389-4a3c-9ed5-cc21b4aafc51">
**Filter with level and file source**
<img width="1380" alt="Filter with level and file source" src="https://github.com/user-attachments/assets/c158d48a-26a4-4d2f-98b6-102f3bd85e41">

jason810496 committed:
Fix Logs/utils

jason810496 committed:
Fix test for Logs/utils

pierrejeambrun commented on airflow/www/static/js/dag/details/taskInstance/Logs/utils.test.tsx:
```diff
@@ -40,6 +40,7 @@ const mockTaskLog = `
 [2022-06-04 00:00:02,010] {taskinstance.py:1548} WARNING - Exporting env vars: AIRFLOW_CTX_DAG_OWNER=*** AIRFLOW_CTX_DAG_ID=test_ui_grid
 [2024-07-01 00:00:02,010] {taskinstance.py:1548} INFO - Url parsing test => "https://apple.com", "https://google.com", https://something.logs/_dashboard/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-1d,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState))))
 `;
+const mockExtraLog = `--------------------------------------------------------------------------------`;
```
Can we add an example with a stack trace for example that has intermediate lines, (typical case where only the first line is considered ERROR, while the entire trace should be).

I don't understand what this `mockExtraLog` is supposed to do.

pierrejeambrun commented on airflow/www/static/js/dag/details/taskInstance/Logs/utils.test.tsx:
```diff
@@ -145,7 +156,8 @@ describe("Test Logs Utils.", () => {
     );
 
-    const lines = parsedLogs!.split("\n");
+    // remove the last line which is empty
+    const lines = parsedLogs!.split("\n").filter((line) => line.length > 0);
```
Why do we need that now ?

pierrejeambrun commented on airflow/www/static/js/dag/details/taskInstance/Logs/utils.ts:
```diff
@@ -87,24 +89,19 @@ export const parseLogs = (
   // Coloring (blue-60 as chakra style, is #0060df) and style such that log group appears like a link
   const logGroupStyle = "color:#0060df;cursor:pointer;font-weight:bold;";
 
+  const regExp = /\[(.*?)\] \{(.*?)\} (.*?) -/;
```
The regexp was slightly updated, can you elaborate please ?

pierrejeambrun reviewed (commented):
Tested, working as expected.

Just a few questions.

I think we should also consider applying the colors to those lines. (only certain lines appear red.)

jason810496 committed:
Fix by Review Comment

- Added red color styling to lines based on the `currentLevel`
- Added comments for new regExp

jason810496 committed:
Refactor Logs/utils test cases

pierrejeambrun reviewed (dismissed):
Looks good thanks.

Some of the questions of the previous review are left un answered. Just for my comprehension, if do you mind giving me a few hints.
